packages/web/integrations/moonpay/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { apiClient } from "@osmosis-labs/utils";
+import { observer } from "mobx-react-lite";
+import dynamic from "next/dynamic";
+import { FunctionComponent, useMemo } from "react";
+
+import { MoonpaySignUrlResponse } from "~/integrations/moonpay/types";
+import { ModalBaseProps } from "~/modals";
+import { useStore } from "~/stores";
+
+const MoonPayBuyWidget = dynamic(
+  () => import("@moonpay/moonpay-react").then((mod) => mod.MoonPayBuyWidget),
+  { ssr: false }
+);
+
+async function generateMoonpayUrlSignature(url: string): Promise<string> {
+  return (
+    await apiClient<MoonpaySignUrlResponse>(
+      "/api/integrations/moonpay/sign-url",
+      {
+        method: "POST",
+        data: {
+          url,
+        },
+      }
+    )
+  ).signature;
+}
+
+export const Moonpay: FunctionComponent<
+  { assetKey: string } & Pick<ModalBaseProps, "isOpen" | "onRequestClose">
+> = observer(({ assetKey, isOpen }) => {
+  const { chainStore, accountStore } = useStore();
+
+  const prefersDark = useMemo(
```

Comment:
We have `react-use` inside our codebase we can use their hook for this media query:
https://streamich.github.io/react-use/?path=/story/sensors-usemedia--docs